cpp/src/gandiva/precompiled/time_test.cc

```diff
@@ -382,6 +386,29 @@ TEST(TestTime, TimeStampAdd) {
   EXPECT_EQ(add_date64_int64(StringToTimestamp("2000-02-27 00:00:00"), 4),
             StringToTimestamp("2000-03-02 00:00:00"));
 
+  EXPECT_EQ(add_timestamp_month_interval(StringToTimestamp("2000-02-27 00:00:00"), 4),
```

Comment:
What happens if the user passes a negative number to this function? It works or raise an error?

Reply:
Another test case is using a huge number to add months, like the limit of the integer

Reply:
Another test case, for leap years, if the date is `2016-01-29` and you add one month, check if the result will be `2016-01-28`

Reply:
Done, I couldn't use the limit for INT32 on month_interval because if I did it it would overflow the timestamp result, since it's adds INT32_MAX*millis_in_a_month, so the result would be to big to int64 to handle. Also I did not added tests for daytime intervals in the type limit because it returns some timestamps too big to make even sense like 185543538537600000